generator_gomod.go

```diff
@@ -54,6 +77,100 @@ func withGomod(g *SourceGenerator, srcSt, worker llb.State, opts ...llb.Constrai
 	}
 }
 
+func withCredHelper(credHelper llb.State, credHelperPath string) RunOptFunc {
+	return func(ei *llb.ExecInfo) {
+		llb.AddMount(credHelperPath, credHelper, llb.SourcePath(GitCredentialHelperGomod)).SetRunOption(ei)
+	}
+}
+
+func (g *SourceGenerator) mountGitAuthConfig(mountPoint, basename string) llb.RunOption {
+	return RunOptFunc(func(ei *llb.ExecInfo) {
+		if g.Gomod == nil || g.Gomod.Auth == nil {
+			return
+		}
+
+		var b bytes.Buffer
+		enc := gob.NewEncoder(&b)
```

Comment:
Why gob?

Reply:
to avoid yaml messiness -- we've already got a go struct, why not keep it that way rather than translating it into another form and back?

Reply:
Mostly because its a weird encoding and is actually less efficient than the stdlib json encoding.

Reply:
Fixed this.

Reply:
By using the filepath to convey the necessary information.